### ISSUE TITLE
Logout breaks isGuest checks somewhere

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#11149]
```
TypeError: Cannot read property 'isGuest' of null
    at t.render (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:41:426009)
    at ya (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:65582)
    at va (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:65377)
    at Zs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:99683)
    at qs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:83963)
    at Gs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:80988)
    at Ms (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:79559)
    at vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41813
    at t.unstable_runWithPriority (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:67:3884)
    at po (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41541)
    at _o (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41758)
    at go (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41693)
    at Pe (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:109447)
    at Un (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:31680)
    at Bn (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:31397)
2019-10-14T14:31:29.069Z E Cannot read property 'isGuest' of null
TypeError: Cannot read property 'isGuest' of null
    at t.render (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:41:426009)
    at ya (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:65582)
    at va (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:65377)
    at Zs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:99683)
    at qs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:83963)
    at Gs (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:80988)
    at Ms (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:79559)
    at vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41813
    at t.unstable_runWithPriority (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:67:3884)
    at po (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41541)
    at _o (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41758)
    at go (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:41693)
    at Pe (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:109447)
    at Un (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:31680)
    at Bn (vector://vector/webapp/bundles/a6f8076a857e57b3741a/bundle.js:59:31397)
2019-10-14T14:31:29.069Z E The above error occured while React was rendering the following components: 
    in RoomTile
    in div
    in t
    in div
    in div
    in t
    in t
    in div
    in RoomSubList
    in div
    in RoomList
    in aside
    in div
    in LeftPanel
    in div
    in t
    in div
    in LoggedInView
    in t
    in MatrixChat
```

**Comment by t3chguy** [comment]
stack points at https://github.com/matrix-org/matrix-react-sdk/blob/f1db0cf027a8dc3c1d3065ae8de40d9beecd50e3/src/components/views/rooms/RoomTile.js#L366

**Comment by jryans** [comment]
The stack above is from the client being `null`, which should be fixed generally by https://github.com/matrix-org/matrix-react-sdk/pull/3533 (which is meant to unmount components first before stopping the client).

Since that PR landed after Riot 1.4.2, I'll consider this a duplicate.
